Client/Ecosia/Extensions/UIView+maskedCorners.swift

```diff
@@ -0,0 +1,34 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+import Foundation
+
+extension UIView {
+    fileprivate struct Positions: OptionSet {
+        static let top = Positions(rawValue: 1)
+        static let bottom = Positions(rawValue: 1 << 1)
+        let rawValue: Int8
+        
+        static func derive(row: Int, totalCount: Int) -> Positions {
+            var pos = Positions()
+            if row == 0 { pos.insert(.top) }
+            if row == totalCount - 1 { pos.insert(.bottom) }
+            return pos
+        }
+    }
+    
+    func setMaskedCornersUsingPosition(row: Int, totalCount: Int) {
```

Comment:
You're right, comments here will definitely help, will add them 👍 

I think in general I didn't stop to think too much about comments this time around, feel free to flag any places you see where it might be needed!